### PR TITLE
DeleteNodeCommand test class

### PR DESCRIPTION
**Pull request by CharlotteXiaoYeDing**
There is code duplication between removeNode(), insertNode() and layout(). 
aNeedsLayout is set to be true twice.

### CODE REVIEW

File: test/ca/mcgill/cs/stg/jetuml/commands/TestDeleteNodeCommand.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+import java.util.ArrayList;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import ca.mcgill.cs.stg.jetuml.diagrams.ClassDiagramGraph;
+import ca.mcgill.cs.stg.jetuml.graph.ClassNode;
+import ca.mcgill.cs.stg.jetuml.graph.Graph;
+import ca.mcgill.cs.stg.jetuml.graph.Node;
+
```
Comment (prmr) [inline test/ca/mcgill/cs/stg/jetuml/commands/TestDeleteNodeCommand.java]: Please activate Checkstyle and use the style file in `/style/Style.xml`

File: test/ca/mcgill/cs/stg/jetuml/commands/TestDeleteNodeCommand.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+public class TestDeleteNodeCommand {
+    private Graph aGraph;
+    private Field aNeedsLayout;
+    private Field aNodesToBeRemoved;
+    private Node aNode;
+    private ArrayList<Node> aRootNodes;
+    private DeleteNodeCommand aDeleteNodeCommand;
+    private boolean hasExecuted = false;
```
Comment (prmr) [inline test/ca/mcgill/cs/stg/jetuml/commands/TestDeleteNodeCommand.java]: `hasExecuted` should be named with the usual convention for fields. More importantly, I don't get how this field is supposed to be set as a result of the command being executed.

File: test/ca/mcgill/cs/stg/jetuml/commands/TestDeleteNodeCommand.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+import ca.mcgill.cs.stg.jetuml.graph.ClassNode;
+import ca.mcgill.cs.stg.jetuml.graph.Graph;
+import ca.mcgill.cs.stg.jetuml.graph.Node;
+
+public class TestDeleteNodeCommand {
+    private Graph aGraph;
+    private Field aNeedsLayout;
+    private Field aNodesToBeRemoved;
+    private Node aNode;
+    private ArrayList<Node> aRootNodes;
```
Comment (prmr) [inline test/ca/mcgill/cs/stg/jetuml/commands/TestDeleteNodeCommand.java]: `aRootNodes` ends up only being an alias for the object inside the graph. I would remove the field and access the value locally when needed.

File: test/ca/mcgill/cs/stg/jetuml/commands/TestDeleteNodeCommand.java [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    private DeleteNodeCommand aDeleteNodeCommand;
+    private boolean hasExecuted = false;
+    
+    @Before
+    public void setUp() throws Exception
```
Comment (prmr) [inline test/ca/mcgill/cs/stg/jetuml/commands/TestDeleteNodeCommand.java]: catch all the relevant exception types in a catch clause that calls `fail()`, it's cleaner.

File: test/ca/mcgill/cs/stg/jetuml/commands/TestDeleteNodeCommand.java [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+        aNodesToBeRemoved = aGraph.getClass().getSuperclass().getDeclaredField("aNodesToBeRemoved");
+        aNodesToBeRemoved.setAccessible(true);
+        aDeleteNodeCommand = new DeleteNodeCommand(aGraph, aNode);  
+    }
+    
+    @Test
+    public void testExecute() {
+        aDeleteNodeCommand.execute();
+        hasExecuted = true;
+        try {
```
Comment (prmr) [inline test/ca/mcgill/cs/stg/jetuml/commands/TestDeleteNodeCommand.java]: You can encapsulate the exception handling in a method. See how it's done in `TestPropertyChangeTracker.getPropertyName`

File: test/ca/mcgill/cs/stg/jetuml/commands/TestDeleteNodeCommand.java [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+        } catch (IllegalArgumentException e) {
+            e.printStackTrace();
+        } catch (IllegalAccessException e) {
+            e.printStackTrace();
+        } 
+    }
+    
+    @Test
+    public void testUndo() {
+        if (!hasExecuted)
```
Comment (prmr) [inline test/ca/mcgill/cs/stg/jetuml/commands/TestDeleteNodeCommand.java]: Tests should be independent

File: test/ca/mcgill/cs/stg/jetuml/commands/TestDeleteNodeCommand.java (hunk 2 above)
Comment (CharlotteXiaoYeDing) [inline test/ca/mcgill/cs/stg/jetuml/commands/TestDeleteNodeCommand.java]: This is attribute is to make sure that we execute() the command before undo() it.

**Comment by CharlotteXiaoYeDing** [comment]
I'll do the same changes for the other tests

File: .checkstyle [hunk 7]
```diff
@@ -1,13 +1,13 @@
-<?xml version="1.0" encoding="UTF-8"?>
```
Comment (prmr) [inline .checkstyle]: Please remove this file from the commit

File: test/ca/mcgill/cs/stg/jetuml/commands/TestDeleteNodeCommand.java [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+        {
+            fail();
+        }
+    }
+
+    @Test
+    public void testUndo() 
+    {
+        aDeleteNodeCommand.execute();
+        int numOfNodes = aGraph.getRootNodes().size();
```
Comment (prmr) [inline test/ca/mcgill/cs/stg/jetuml/commands/TestDeleteNodeCommand.java]: Just to be explicit you can put in `assertEquals(0, aGraph.getRootNodes().size());`

**Comment by prmr** [comment]
You can remove the statement `aGraph.layout();` in both `DeleteNodeCommand.execute()` and `DeleteNodeCommand.undo()`, because they are indeed redundant.

**Comment by CharlotteXiaoYeDing** [comment]
Hi Martin,

I’ve deleted .checkstyle and removed layout() in DeleteNodeCommand. Do I need to remove layout() in other commands?

**Pull request closed without merging**